fuel-vm/src/interpreter/executors/instruction.rs

```diff
@@ -759,7 +759,8 @@ where
 
             Instruction::LDC(ldc) => {
                 let (a, b, c) = ldc.unpack();
-                self.dependent_gas_charge(self.gas_costs().ldc, r!(c))?;
+                let contract_size = self.load_contract_size(r!(a))?;
+                self.dependent_gas_charge(self.gas_costs().ldc, r!(contract_size))?;
                 self.load_contract_code(r!(a), r!(b), r!(c))?;
```

Comment:
If I understand correctly, the new approach is:
- Retrieve the contract size (in bytes) of the given contract (`$rA`) from storage
- Instead of charging for the user-requested size (potentially 0), we now charge for the amount of contract bytes retrieved from storage
- Finally, as before, load the contract code

Essentially, we now charge for the full size of the contract, regardless of how much is requested.

It seems like this approach makes sense given that we load contracts in their entirety:
```rust
pub(crate) fn contract<'s, S>(
    storage: &'s S,
    contract: &ContractId,
) -> ...
where
    S: InterpreterStorage,
{
    storage
        .storage_contract(contract)
    ...
}

``` 
rather than the requested size, anyway.

Have I understood this correctly?

Reply:
That sounds all correct. I'm simplifying some of this code right now to get the last test to pass.